### PR TITLE
IP materials

### PR DESCRIPTION
**Pull request by skorpiom**
materials routes with auth, just admin can add materials
gonna add tests later, just pls check funcionality 

get /materials - get all materials
get /materials/:id - get material with id
post /materials/section/:id - create material in section with id
patch  /materials/:id - update material with id
delete /materials/:id/section/:sectionID - delete material with id from section with id

### CODE REVIEW

File: Api/Src/Services/MaterialService.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import * as express from "express";
+import * as mongoose from "mongoose";
+
+import MaterialRepository from "../Repositories/MaterialRepository";
+import MaterialSchema from "../Models/Material";
+import GradeSchema from "../Models/Grade";
```
Comment (robert-czwartosz) [inline Api/Src/Services/MaterialService.ts:6]: Grade schema is not needed

File: Api/Src/Services/MaterialService.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+export default class MaterialService {
+
+    constructor(private repository: MaterialRepository, private sectionService: SectionService) {}
+
+    createMaterial = async ( req: express.Request ) => {
+        const id = new mongoose.Types.ObjectId(req.params.id);
+        const material = new MaterialSchema(req.body);
+        await material.validate();
+
+        let section: TSection = await this.sectionService.getSectionById(id);
```
Comment (robert-czwartosz) [inline Api/Src/Services/MaterialService.ts:20]: if (!section) throw Error("Section not found");

File: Api/Src/Services/MaterialService.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    getMaterialById = async ( req: express.Request ) => {
+        const id = new mongoose.Types.ObjectId(req.params.id);
+        return this.repository.getById(id);
+    }
+
+    updateMaterial = async ( req: express.Request ) => {
+        const id = new mongoose.Types.ObjectId(req.params.id);
+        const material = new GradeSchema(req.body);
```
Comment (robert-czwartosz) [inline Api/Src/Services/MaterialService.ts:39]: GradeSchema --> MaterialSchema

File: Api/Src/Services/MaterialService.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        material._id = id;
+        await material.validate();
+        await this.repository.updateById(id, material);
+        return material;
```
Comment (robert-czwartosz) [inline Api/Src/Services/MaterialService.ts:43]: const result = await this.repository.updateById(id, material);
return result;

File: Api/Src/Services/MaterialService.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    deleteMaterial = async ( req: express.Request ) => {
+        const idMaterial = new mongoose.Types.ObjectId(req.params.id);
+        const idSection = new mongoose.Types.ObjectId(req.params.sectionID);
+
+        let section: TSection = await this.sectionService.getSectionById(idSection);
```
Comment (robert-czwartosz) [inline Api/Src/Services/MaterialService.ts:50]: if (!section) throw Error("Section not found");

File: Api/Src/Services/MaterialService.ts (hunk 1 above)
Comment (skorpiom) [reply Api/Src/Services/MaterialService.ts]: sure

File: Api/Src/Services/MaterialService.ts (hunk 2 above)
Comment (skorpiom) [reply Api/Src/Services/MaterialService.ts:19]: added

File: Api/Src/Services/MaterialService.ts (hunk 3 above)
Comment (skorpiom) [reply Api/Src/Services/MaterialService.ts]: changed

File: Api/Src/Services/MaterialService.ts (hunk 4 above)
Comment (skorpiom) [reply Api/Src/Services/MaterialService.ts]: changed

File: Api/Src/Services/MaterialService.ts (hunk 5 above)
Comment (skorpiom) [reply Api/Src/Services/MaterialService.ts:50]: added

**Review by robert-czwartosz — APPROVED** [review]

**Review by fszwedo — APPROVED** [review]

**Pull request merged**